src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java

```diff
@@ -994,10 +983,6 @@ private Object readResolve() {
         if (podLabels == null && labels != null) {
             setPodLabels(PodLabel.fromMap(labels));
         }
-        // if unset, should default to no limit
-        if (containerCap == null) {
-            containerCap = Integer.MAX_VALUE;
-        }
```

Comment:
IIUC we need a `readResolve` which sets `0` (old "unlimited") to `null` (new "unlimited").

Once `null`, the getter will convert `null` to `Integer.MAX_VALUE`.

Correct?

Reply:
https://github.com/jenkinsci/kubernetes-plugin/pull/865

Reply:
Disregard. https://github.com/jenkinsci/kubernetes-plugin/pull/866 is the only way.